src/lib/ext-activitylog.js

```diff
@@ -0,0 +1,143 @@
+import { save, load } from '../lib/save-load.js';
+
+export default class ActivityLog {
+  constructor() {
+    this.logs = [];
+    this.loadedLogs = [];
+    this.rowIndex = 0;
+
+    this.loadLogBtn = document.getElementById('loadLogBtn');
+    this.logInputFile = document.getElementById('logInputFile');
+    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.showLogDetailWrapper = document.querySelector(
+      '.show-log-detail-wrapper'
+    );
+    this.showLogDetails = document.getElementById('showLogDetails');
+    this.showLoadedLog = document.getElementById('showLoadedLog');
+    this.logTableWrapper = document.querySelector('.log-table');
+    this.table = document.querySelector('table');
+    this.notice = document.querySelector('.notice');
+    this.closeBtn = document.querySelector('.close');
+  }
+
+  async getExistingLogs() {
+    const { existingLogs } = await browser.runtime.sendMessage({
+      requestType: 'sendAllLogs',
+      requestTo: 'ext-monitor',
+    });
+    return existingLogs;
+  }
+
+  async loadFile(file) {
+    try {
+      const loadedLogsAsText = await load.loadLogAsText(file);
+      const loadedJSONLogs = JSON.parse(loadedLogsAsText);
+      this.loadedLogs.push(...loadedJSONLogs);
+      // TODO: Display loaded logs in table view
+
+      if (this.notice.textContent.trim().length) {
+        this.notice.textContent = '';
+        this.notice.classList.remove('failure');
+      }
+    } catch (error) {
+      this.notice.textContent = error.message;
+      this.notice.classList.add('failure');
+    }
+  }
+
+  async saveLogs(logs) {
```

Comment:
Nit, in my opinion this method should be only responsible for saving the logs that are stored in `this.logs` and so it would not need to get the array of logs as a parameter, can just use `this.logs` internally.

Reply:
Yeah, I have updated this method excluding the parameter and using `this.logs` internally, but in the future when we may need to save logs after filtering the logs in the table, it could be helpful if we pass the filtered logs as an argument. That was my thinking of having a parameter here.